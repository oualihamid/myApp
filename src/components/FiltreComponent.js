import React, { useState } from "react";
import { Modal, StyleSheet, Pressable, View, ScrollView } from "react-native";
import { Card,Title,Paragraph,Badge,Chip,Divider,TextInput,Switch,Button    } from 'react-native-paper';

const plusSCreen = ({data,setData }) => {
    const [modalVisible, setModalVisible] = useState(false);
    const [name, setName] = useState("");
    const [open, setOpen] = useState(true);
    const [nbBike, setNbBike] = useState("");

    let newArray = data.filter(function (el) {
      let status = open == true ? "OPEN":"CLOSE";    
      return el.available_bikes >= nbBike &&  el.status == status &&  el.name.toUpperCase().includes(name.toUpperCase()); 
    });

  const onToggleSwitch = () => setOpen(!open);
  return (
    <View>
      <Modal
        animationType="slide"
        transparent={true}
        visible={modalVisible}
        onRequestClose={() => {
          setModalVisible(!modalVisible);
        }}
      >
        <View >
          <View style={styles.modalView}>
            <Pressable
                onPress={() => setModalVisible(!modalVisible)}
              >  
              <Badge size={45} style={styles.close}>X</Badge>
            </Pressable>  
            <ScrollView>
                    <Card style={styles.item}>
                        <Card.Content>
                          <Title>Filtrer le r√©sultat</Title>
                          <Paragraph>Nom de la station</Paragraph>
                           <TextInput
                            label="Nom"
                            value={name}
                            onChangeText={text => setName(text)}
                          />  
                          <Paragraph style={styles.title2}>Ouvert </Paragraph>
                          <Switch value={open} onValueChange={onToggleSwitch}/>

                          <Paragraph style={styles.title2}>Nb de Bike disponible</Paragraph>
                           <TextInput
                            label="Taper un nombre"
                            value={nbBike}
                             keyboardType="numeric"
                            onChangeText={nb => setNbBike(nb)}
                          />
                          <Button  style={styles.buttonShow} mode="contained" onPress={() =>{setData(newArray); setModalVisible(!modalVisible)} }>
                            Afficher ({newArray.length})
                          </Button>  

                      </Card.Content>
                    </Card>
                </ScrollView>  
          </View>
        </View>
      </Modal>
      <Pressable onPress={() => setModalVisible(true)}>  
        <Badge size={45} style={styles.button}>FILTRER</Badge>
      </Pressable> 
    </View>
  );
};

const styles = StyleSheet.create({
  modalView: {
    width:'100%',
    height:'100%',
    //top:10,
    //margin:40,
    backgroundColor: "white",
    borderRadius: 20,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5
  },
  button: {
    width:120,
    color:'#FFF',
    borderRadius:20,
    fontSize:14,
    fontWeight:'bold',
    backgroundColor:'#C0C0C0', 
    margin:5,
  },
  close:{
    margin:10,  
    backgroundColor:'#C0C0C0',
    color:'#FFF'
  },
  buttonShow:{
    marginTop:20,
    backgroundColor:'#ff0078'
  }

});

export default plusSCreen;