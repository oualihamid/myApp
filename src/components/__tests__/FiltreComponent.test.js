import React from 'react';
import renderer from 'react-test-renderer';
import FiltreComponent from '../FiltreComponent';

let stations=[
  {
    "number": 55,
    "contract_name": "toulouse",
    "name": "00055 - SAINT-SERNIN - G. ARNOULT",
    "address": "2 RUE GATIEN ARNOULT",
    "position": {
      "lat": 43.6089519604964,
      "lng": 1.4410035987262
    },
    "banking": true,
    "bonus": false,
    "bike_stands": 15,
    "available_bike_stands": 15,
    "available_bikes": 0,
    "status": "OPEN",
    "last_update": 1653222594000
  },
  {
    "number": 195,
    "contract_name": "toulouse",
    "name": "00195 - LARROUSSE - RIEUX",
    "address": "FACE 15 RUE PIERRE LAROUSSE",
    "position": {
      "lat": 43.5972354030358,
      "lng": 1.45907112459247
    },
    "banking": true,
    "bonus": false,
    "bike_stands": 18,
    "available_bike_stands": 16,
    "available_bikes": 2,
    "status": "OPEN",
    "last_update": 1653222777000
  }
];

it('renders correctly', () => {
  const tree = renderer.create(<FiltreComponent data={stations} />).toJSON();
  expect(tree).toMatchSnapshot();
});
