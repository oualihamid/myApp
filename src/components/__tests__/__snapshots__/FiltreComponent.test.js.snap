// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      collapsable={false}
      numberOfLines={1}
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#C0C0C0",
          "borderRadius": 20,
          "color": "#FFF",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "bold",
          "height": 45,
          "lineHeight": 45,
          "margin": 5,
          "minWidth": 45,
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 4,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": 120,
        }
      }
    >
      FILTRER
    </Text>
  </View>
</View>
`;
